app/services/acuant/request.rb

```diff
@@ -0,0 +1,68 @@
+module Acuant
+  class Request
+    def path
+      raise NotImplementedError
+    end
+
+    def body
+      raise NotImplementedError
+    end
+
+    def handle_http_response(_response)
+      raise NotImplementedError
+    end
+
+    def method
+      :get
+    end
+
+    def url
+      URI.join(Figaro.env.acuant_assure_id_url, path)
+    end
+
+    def headers
+      {
+        'Authorization' => authorization_header_value,
+        'Accept' => 'application/json',
+      }
+    end
+
+    def send
+      http_response = send_http_request
+      return handle_invalid_response(http_response) unless http_response.success?
+
+      handle_http_response(http_response)
+    rescue StandardError => e
+      Response.new(success: false, exception: e)
+    end
+
+    private
+
+    def send_http_request
+      # TODO: Timeouts
+      case method.downcase.to_sym
+      when :post
+        Faraday.post(url, body, headers)
+      when :get
+        Faraday.get(url, nil, headers)
+      end
+    end
+
+    def handle_invalid_response(http_response)
+      message = [
+        self.class.name,
+        'Unexpected HTTP response',
+        http_response.status,
+      ].join(' ')
+      exception = RuntimeError.new(message)
+      Response.new(success: false, exception: exception)
+    end
+
+    def authorization_header_value
+      username = Figaro.env.acuant_assure_id_username
+      password = Figaro.env.acuant_assure_id_password
+      digest = Base64.strict_encode64("#{username}:#{password}")
+      "Basic #{digest}"
```

Comment:
Looks like Faraday has a `basic_auth` method so we could use that instead of building this header manually:

https://github.com/lostisland/faraday/issues/426#issuecomment-86116243

Reply:
Yep, meant to come back and clean this up